fvm/src/kernel/default.rs

```diff
@@ -377,23 +379,13 @@ where
     C: CallManager,
 {
     fn total_fil_circ_supply(&self) -> Result<TokenAmount> {
-        self.call_manager
-            .context()
-            .base_circ_supply
-            .checked_add(&self.get_reserve_disbursed()?)
-            .ok_or(anyhow!(
-                "overflow when adding reserve to base circulating supply"
-            ))
-            .or_fatal()?
-            .checked_sub(&self.get_burnt_funds()?)
-            .ok_or(anyhow!("underflow when subtracting burnt funds"))
-            .or_fatal()?
-            .checked_sub(&self.power_locked()?)
-            .ok_or(anyhow!("underflow when subtracting power locked funds"))
-            .or_fatal()?
-            .checked_sub(&self.market_locked()?)
-            .ok_or(anyhow!("underflow when subtracting market locked funds"))
-            .or_fatal()
+        Ok((&self.call_manager.context().fil_vested
+            + &self.get_fil_mined()?
+            + &self.get_reserve_disbursed()?
+            - &self.get_burnt_funds()?
+            - &self.power_locked()?
+            - &self.market_locked()?)
```

Comment:
Can we add `map_err` here to add a context to these errors, like previously? Although probably not worth it because these errors will actually never happen.

Reply:
Previously, we were adding context to overflow errors (which couldn't happen), not the other errors. But I'll go ahead and change the actual functions to add context to their errors.